crates/types/src/nodes_config.rs

```diff
@@ -325,10 +325,21 @@ pub enum StorageState {
     ///
     /// The node can never transition back to `Provisioning` once it has transitioned out of it.
     ///
-    /// should read from: no
-    /// can write to: no
+    /// [authoritative]
+    /// or all intents and purposes, this is equivalent to a `ReadOnly` state, except that it's
+    /// excluded from nodeset generation. The difference
+    /// between this and `ReadOnly` is that if a node is in Provisioning state, we are confident
+    /// that it has not been added to nodesets and we can safely remove it from the cluster without
+    /// checking the log-chain or trim points of loglets. Note that if this node happens to be in a
```

Comment:
In practice, wouldn't we move the state of the node to `Disabled` before removing it from the cluster to avoid that we don't switch from `Provisioning` to `ReadWrite` while we are removing the node?

Reply:
If the node is physically down (never started), it's safe to put it directly into Disabled, or removing it directly from nodes_config (by transitioning it into a tombstone). Once the node is in tombstone, it's disabled, permanently.

The notion of "remove from the cluster" in this context only refer to using nodes_config to record this node id as tombstone, we should _never_ allow a node-id to actually disappear from the nodes config.